src/lib/core/TLVReader.cpp

```diff
@@ -1023,8 +1023,26 @@ CHIP_ERROR TLVReader::GetElementHeadLength(uint8_t & elemHeadBytes) const
 TLVElementType TLVReader::ElementType() const
 {
     if (mControlByte == static_cast<uint16_t>(kTLVControlByte_NotSpecified))
+    {
+        return TLVElementType::NotSpecified;
+    }
+    int8_t elementType = mControlByte & kTLVTypeMask;
+
+    // Range check
+    if (elementType > static_cast<int8_t>(TLVElementType::EndOfContainer))
+    {
+        // this is not a valid TLV ...
         return TLVElementType::NotSpecified;
-    return static_cast<TLVElementType>(mControlByte & kTLVTypeMask);
+    }
+    if (elementType < static_cast<int8_t>(TLVElementType::Int8))
+    {
+        // this is not a valid TLV ...
+        return TLVElementType::NotSpecified;
```

Comment:
Is NotSpecified the correct error? If so, can we add tests for this in TLVReader, or is it already covered?

Reply:
I do not actually know. Previous code was just casting to an invalid/impossible value. I now return something. I considered chipDie() however that sounds not ideal (we have TLV data encoded in BLE advertisements, so then someone could craft random TLV packets to be sent to us so die sounds wrong). Ideally we would CHIP_ERROR, however this API seems to assume (wrongly?) error is impossible.

Reply:
Also added test data to fuzzing tests. Although reading through those tests they are exceptionally poor: they just test that **some** error happens and not the exact content of the error (so it may be "end of stream" or "unclosed structure"). So my only coverage is "we do not crash" + existing tests.

My change makes things better or the same as before at least.